boefjes/boefjes/plugins/kat_withaname/normalize.py

```diff
@@ -0,0 +1,72 @@
+import json
+from collections.abc import Iterable
+
+import structlog
+
+from octopoes.models import OOI, Reference
+from octopoes.models.ooi.findings import Finding, KATFindingType
+
+logger = structlog.getLogger(__name__)
+
+
+def run(input_ooi: dict[str, str], raw: bytes) -> Iterable[OOI]:
+    """Normalize witha.name output."""
+    result = json.loads(raw)
+
+    if result["targets"] is None:
+        logger.info("No targets found in witha.name output")
+        return
+
+    input_ooi_reference = Reference.from_str(input_ooi["primary_key"])
+    found_targets = None
+
+    ooi_category = "IP" if input_ooi["primary_key"].startswith("IPAddress") else "Hostname"
+
+    if ooi_category == "IP":
+        logger.debug("Found IP address: %s", input_ooi["primary_key"])
+        found_targets = [
+            target for target in result["targets"] if target["ip"] == input_ooi_reference.tokenized.address
+        ]
+    elif ooi_category == "Hostname":
+        logger.debug("Found Hostname address: %s", input_ooi["primary_key"])
+        # Make host always start with www.
+        raw_host = input_ooi["name"].lower()
+        host = raw_host if raw_host.startswith("www.") else "www." + raw_host
+
+        # Because hosts are listed with and without www. prefix, we need to check both
+        found_targets = [target for target in result["targets"] if target["host"] in [host, host.removeprefix("www.")]]
+
+    logger.info("Found targets: %s", found_targets)
+    if found_targets is None:
+        raise ValueError(f"OOI: {input_ooi['primary_key']} is not a valid IP address or hostname.")
+    if not found_targets:
+        logger.info("No targets found for %s", input_ooi["primary_key"])
+        return
```

Comment:
Imo I'd just check `if not found_targets`, since the only relevant error at this stage is missing targets. The first error (invalid input) should already be handled earlier before this normalizer is reached

Reply:
Fixed in 881e0c5